spec/security.js

```diff
@@ -24,11 +37,55 @@ describe('security issues', function() {
                 }
             });
 
-            shouldCompileTo('{{#with this}}{{this.abc}}{{/with}}',
-                new TestClass(), 'xyz');
-            shouldCompileTo('{{#with this}}{{lookup this "abc"}}{{/with}}',
-                new TestClass(), 'xyz');
+
+            expectTemplate('{{#with this}}{{this.abc}}{{/with}}')
+              .withCompileOptions(compileOptions)
+              .withInput(new TestClass())
+              .toCompileTo('xyz');
+
+            expectTemplate('{{#with this}}{{lookup this "abc"}}{{/with}}')
+              .withCompileOptions(compileOptions)
+              .withInput(new TestClass())
+              .toCompileTo('xyz');
         });
+
+        it('should not allow constructors to be accessed', function() {
+          expectTemplate('{{lookup (lookup this "constructor") "name"}}')
+            .withCompileOptions(compileOptions)
+            .withInput({})
+            .toCompileTo('');
+          if (compileOptions.strict) {
+            expectTemplate('{{constructor.name}}')
+              .withCompileOptions(compileOptions)
+              .withInput({})
+              .toThrow(TypeError);
```

Comment:
This was already the case in 3.x before this change and remains after this change currently.   `strict` mode resulted in a `TypeError` when accessing `constructor`, `__proto__`, etc.  We can fix this but it would require more changes.